Flow.Launcher/SettingPages/Views/SettingsPaneTheme.xaml

```diff
@@ -24,109 +24,368 @@
         </ResourceDictionary>
     </ui:Page.Resources>
     <ScrollViewer
-        Padding="6,0,24,0"
-        CanContentScroll="True"
+        Padding="6 0 24 0"
+        CanContentScroll="False"
         FontSize="14"
         VirtualizingStackPanel.IsVirtualizing="True"
         VirtualizingStackPanel.ScrollUnit="Pixel">
         <StackPanel>
             <!--  Page title  -->
             <TextBlock
-                Margin="5,23,0,5"
+                Margin="5 23 0 5"
                 FontSize="30"
                 Style="{StaticResource PageTitle}"
                 Text="{DynamicResource appearance}"
                 TextAlignment="left"
                 Visibility="Collapsed" />
+            <!--  Theme Preview and Editor  -->
+            <Grid Height="380">
+                <Grid.Resources>
+                    <!--  Style for the first ColumnDefinition  -->
+                    <Style x:Key="FirstColumnStyle" TargetType="ColumnDefinition">
+                        <Style.Triggers>
+                            <DataTrigger Binding="{Binding ElementName=Editor, Path=IsChecked}" Value="False">
+                                <Setter Property="Width" Value="*" />
+                            </DataTrigger>
+                            <DataTrigger Binding="{Binding ElementName=Editor, Path=IsChecked}" Value="True">
+                                <Setter Property="Width" Value="8*" />
+                            </DataTrigger>
+                        </Style.Triggers>
+                    </Style>
+                    <!--  Style for the second ColumnDefinition  -->
+                    <Style x:Key="SecondColumnStyle" TargetType="ColumnDefinition">
+                        <Style.Triggers>
+                            <DataTrigger Binding="{Binding ElementName=Editor, Path=IsChecked}" Value="False">
+                                <Setter Property="Width" Value="0" />
+                            </DataTrigger>
+                            <DataTrigger Binding="{Binding ElementName=Editor, Path=IsChecked}" Value="True">
+                                <Setter Property="Width" Value="2*" />
+                                <Setter Property="MinWidth" Value="160" />
+                            </DataTrigger>
+                        </Style.Triggers>
+                    </Style>
+                </Grid.Resources>
+                <Grid.ColumnDefinitions>
+                    <ColumnDefinition MinWidth="500" Style="{StaticResource FirstColumnStyle}" />
+                    <ColumnDefinition Width="0" />
+                    <ColumnDefinition MaxWidth="250" Style="{StaticResource SecondColumnStyle}" />
+                </Grid.ColumnDefinitions>
+                <GridSplitter
+                    Grid.Column="1"
+                    Width="0"
+                    HorizontalAlignment="Stretch"
+                    Background="Transparent" />
```

Comment:
Initially, the code was a bit short. I tried to use the gridsplitter like the previewarea resize, but it would ignore the style trigger, so I disabled it. Apart from that issue, but this layout can designed to auto resize to the length of the label (if the label is very long in certain languages).